src/mbpfan.c

```diff
@@ -524,6 +529,12 @@ void retrieve_settings(const char* settings_path, t_fans* fans)
                 polling_interval = result;
             }
 
+            result = settings_get_int(settings, "general", "intel_pstate_control");
+
+            if (result != 0) {
+                intel_pstate_control = (result == 0) ? 0 : 1;
```

Comment:
Isn't this always 1?  So you can just set `intel_pstate_control = result`.

Reply:
Would there be a case where it's not in the config file? Anyone upgrading from a previous release may not immediately update their config file.

Reply:
Yes we must handle the missing balue case as well.  But it appears that `settings_get_int` returns 0 when it is missing, the same as disabled?